analyses/cms-open-data-ttbar/jetassignment_training.py

```diff
@@ -0,0 +1,842 @@
+
+        # load model
+        best_model = mlflow.xgboost.load_model(best_model_path)
+    
+        # register best model in mlflow model repository
+        if MODEL_REGISTERING:
+            mlflow.register_model(best_model_path, utils.config_training["ml"]["MODEL_NAME"])
+            
+    else:
+        best_model = res[np.argmax(scores)]["full_result"]["model"]
+        
+    # save models locally
+    if even:
+        best_model.save_model(f"models/model_{datetime.datetime.today().strftime('%y%m%d')}_even.json")
+    else:
+        best_model.save_model(f"models/model_{datetime.datetime.today().strftime('%y%m%d')}_odd.json")
+        
+    return best_model
+
+
+# %% tags=[]
+# train on even events
+res = run_training(features_even, labels_even, samples_even, evaluation_matrix, 
+                   N_EVENTS_TRAIN, USE_DASK_ML, USE_MLFLOW, MODEL_LOGGING)
+# save model
+best_model_even = save_register_models(res, USE_MLFLOW, MODEL_LOGGING, MODEL_REGISTERING, even=True)
+
+# train on odd events
+res = run_training(features_odd, labels_odd, samples_odd, evaluation_matrix, 
+                   N_EVENTS_TRAIN, USE_DASK_ML, USE_MLFLOW, MODEL_LOGGING)
+# save model
+best_model_odd = save_register_models(res, USE_MLFLOW, MODEL_LOGGING, MODEL_REGISTERING, even=False)
+
+# %% [markdown]
+# # Evaluation with Optimized Model
+
+# %% tags=[]
+# generate text for NVIDIA Triton config file
+config_txt = utils.ml.write_triton_config(utils.config_training["ml"]["MODEL_NAME"], 
+                                          20, # number of features
+                                          backend_name = "fil", 
+                                          model_type = "xgboost", 
+                                          max_batch_size = 50000000, 
+                                          predict_proba = "true")
+
+print(config_txt)
+
+# %% tags=[]
+# !mkdir reconstruction_bdt_xgb
+
+# %% tags=[]
+with open('reconstruction_bdt_xgb/config.pbtxt', 'w') as the_file:
+    the_file.write(config_txt)
+
+# %% tags=[]
+# !mkdir reconstruction_bdt_xgb/1
+
+# %% tags=[]
+best_model_even.save_model("reconstruction_bdt_xgb/1/xgboost.model")
+
+# %% tags=[]
+# !mkdir reconstruction_bdt_xgb/2
+
+# %% tags=[]
+best_model_even.save_model("reconstruction_bdt_xgb/2/xgboost.model")
+
+# %% tags=[]
+# check out the model directory
+for path, subdirs, files in os.walk("reconstruction_bdt_xgb"):
+    for name in files:
+        print(os.path.join(path, name))
+
+# %% [markdown]
+# If you are using UNL open data, you can upload the model repository to the Triton server using `mc` in the command line:
+#
+# ```
+# mc alias set triton http://$BUCKET_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY
+# mc cp -r reconstruction_bdt_xgb triton/$BUCKET_NAME/reconstruction_bdt_xgb/
+# ```
+#
+# The server may need to be restarted in order to load the model.
+
+# %% tags=[]
+# remove model directory after uploading to triton
+# !rm -r reconstruction_bdt_xgb
```

Comment:
It would be good to remove this feature by default to allow the model files to be still available after running the notebook. Perhaps we can delete the folder before creating new files to avoid any kind of collisions, but even that is maybe optional.